packages/base-types/src/request/index.ts

```diff
@@ -80,9 +80,20 @@ export interface IntentRequest extends BaseRequest {
   payload: IntentRequestPayload;
 }
 
-export interface GeneralRequest extends BaseRequest {
+export interface GeneralUnknownRequest extends BaseRequest {
   type: string; // the general request type is dynamic
-  payload?: ActionAndLabelRequestPayload;
+  payload?: unknown;
+}
+
+export interface GeneralActionAndLabelRequest extends GeneralUnknownRequest {
+  payload: ActionAndLabelRequestPayload;
+}
+
+export type GeneralRequest = GeneralUnknownRequest | GeneralActionAndLabelRequest;
```

Comment:
Split `GeneralRequest` into two subtypes, `GeneralUnknownRequest` and `GeneralActionAndLabelRequest` where the former type reflects the actual request data we are now accounting for.